### PR TITLE
[KOGITO-8676] Adding OnOverflow annotation

### PR DESCRIPTION
**Pull request by fjtirado**
As per [JIRA](https://issues.redhat.com/browse/KOGITO-8676) 
See modified example at https://github.com/kiegroup/kogito-examples/pull/1610
Update doc: https://github.com/kiegroup/kogito-docs/pull/315

### CODE REVIEW

**Review by ricardozanini — APPROVED** [review]
Nice!

File: quarkus/addons/messaging/deployment/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/deployment/ChannelMappingStrategy.java [hunk 1]
```diff
@@ -89,7 +91,15 @@ private static ChannelInfo getChannelInfo(String property, String prefix, boolea
         String name = property.substring(prefix.length(), property.lastIndexOf('.'));
         return new ChannelInfo(name, triggers.getOrDefault(name, Collections.singleton(name)),
                 getClassName(config.getOptionalValue(getPropertyName(prefix, name, "value." + (isInput ? "deserializer" : "serializer")), String.class)), isInput,
-                name.equals(defaultChannelName), config.getOptionalValue((isInput ? UNMARSHALLLER_PREFIX : MARSHALLER_PREFIX) + name, String.class));
+                name.equals(defaultChannelName), config.getOptionalValue((isInput ? UNMARSHALLLER_PREFIX : MARSHALLER_PREFIX) + name, String.class),
+                isInput ? Optional.empty() : onOverflowInfo(name));
+    }
+
+    private static Optional<OnOverflowInfo> onOverflowInfo(String name) {
+        final String namePrefix = KOGITO_EMITTER_PREFIX + name + ".";
```
Comment (MarianMacik) [inline quarkus/addons/messaging/deployment/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/deployment/ChannelMappingStrategy.java:99]: Global config would be also nice (as said in the JIRA, already discussed on chat) + would you mind updating docs with these new config options?

Comment (fjtirado) [reply quarkus/addons/messaging/deployment/src/main/java/org/kie/kogito/addon/cloudevents/quarkus/deployment/ChannelMappingStrategy.java:101]: https://github.com/kiegroup/kogito-docs/pull/315

**Comment by fjtirado** [comment]
> Nice! I'm assuming you will update the guides? (:

https://github.com/kiegroup/kogito-docs/pull/315

**Review by MarianMacik — APPROVED** [review]

**Pull request merged**